public/app/features/dashboard-scene/settings/annotations/AnnotationSettingsEdit.tsx

```diff
@@ -183,6 +184,14 @@ export const AnnotationSettingsEdit = ({ annotation, editIndex, panels, onUpdate
         <Field label="Data source" htmlFor="data-source-picker">
           <DataSourcePicker annotations variables current={annotation.datasource} onChange={onDataSourceChange} />
         </Field>
+        {!ds ||
+          (!ds.annotations && (
```

Comment:
should this not check meta.annotations as well? 

{ds?.meta.annotations && (

Reply:
indeed, I keep forgetting what you mentioned regarding old data sources 🙈